Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -17,6 +17,100 @@ import NIO
 let crlf: StaticString = "\r\n"
 let headerSeparator: StaticString = ": "
 
+/* private but tests */ internal struct HTTPNonContiguousHeaderCommaSeparatedValuedFieldHandler
+: IteratorProtocol {
+    
+    typealias Element = ByteBuffer
+    
+    fileprivate static let comma       = UInt8(",".utf8CString[0])
+    fileprivate static let whiteSpaces = [" ", "\t"]
+                                         .map( {UInt8($0.utf8CString[0])} )
+    
+    private var isNowOnAField = false
+    private var currentHeaderIndex: Int = -1
+    private var singleValueParser: ByteBufferSliceSplitIterator
+    private var byteBuffer: ByteBuffer
+    private let headerName: ContiguousArray<UInt8>
+    private let headers: [HTTPHeader]
+
+    mutating func next() -> ByteBuffer? {
+        // Still not reading a specific field
+        if !isNowOnAField {
+            // Try to get next match
+            currentHeaderIndex += 1
+            while currentHeaderIndex < headers.count {
+                let currentHeader = headers[currentHeaderIndex]
+                if byteBuffer.getSlice(at: currentHeader.name.start,
+                                       length: currentHeader.name.length)?
+                    .compareReadingToCaseInsensitiveCString(headerName) ?? false {
+                    // Use this index
+                    singleValueParser = ByteBufferSliceSplitIterator(
+                        byteBuffer: byteBuffer,
+                        separator: HTTPNonContiguousHeaderCommaSeparatedValuedFieldHandler.comma,
+                        start: currentHeader.value.start,
+                        length: currentHeader.value.length)
+                    isNowOnAField = true
+                    break
+                }
+                currentHeaderIndex += 1
+            }
+        }
+        // Current index is out of bounds = the iterator ends
+        if currentHeaderIndex >= headers.count { return nil }
+        
+        let next = singleValueParser.next()
+        // Null or empty buffer?
+        if (next?.readableBytes ?? 0) == 0 {
+            // Try again
+            isNowOnAField = false
+            return self.next()
+        }
+        return next?.sliceByTrimmingWhitespaces()
+    }
+    
+    init(fromHeaderName headerName: ContiguousArray<UInt8>,
+         fromHeaders: [HTTPHeader],
+         in byteBuffer: ByteBuffer) {
+        
+        singleValueParser = ByteBufferSliceSplitIterator(
+            byteBuffer: byteBuffer,
+            separator: HTTPNonContiguousHeaderCommaSeparatedValuedFieldHandler.comma)
+        headers = fromHeaders
+        self.headerName = headerName
+        
+        self.byteBuffer = byteBuffer
+    }
+
+}
+
+/* private but tests */ internal struct HTTPKeepAliveHandler {
```

Comment:
I don't think a struct for this is very useful, I'd just make this method an internal extension on `HTTPHeaders`.